llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -162,18 +174,18 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineSetExtract(
     break;
   }
   case Intrinsic::aie2_set_I512_I256: {
-    if (cast<GIntrinsic>(*ExtOp).getIntrinsicID() !=
-        Intrinsic::aie2_ext_I256_I512)
-      return false;
-    const Register SetOpIdxReg = MI.getOperand(3).getReg();
-    const Register ExtOpIdxReg = ExtOp->getOperand(3).getReg();
-    auto SetOpCst = getIConstantVRegValWithLookThrough(SetOpIdxReg, MRI);
-    auto ExtOpCst = getIConstantVRegValWithLookThrough(ExtOpIdxReg, MRI);
-    if (SetOpIdxReg != ExtOpIdxReg &&
-        (!SetOpCst || !ExtOpCst ||
-         SetOpCst->Value.getZExtValue() != ExtOpCst->Value.getZExtValue()))
-      return false;
-    break;
+    if (cast<GIntrinsic>(*ExtOp).getIntrinsicID() ==
+            Intrinsic::aie2_ext_I256_I512 &&
```

Comment:
nit: I think a utility like isIntrinsicID(ExtOp, Intrinsic::aie2_ext_I256_I512) would help readability

Reply:
Existing `isIntrinsicID` checks whether the MachineOperand is IntrinsicID or not.
Do I need to define new function for our use case?